build/changelog/ChangelogBuilder.php

```diff
@@ -0,0 +1,149 @@
+<?php
+namespace Aws\Build\Changelog;
+
+/**
+ * @internal
+ */
+
+class ChangelogBuilder 
+{
+  
+  public $newServiceFlag = False;
+
+  public function readChangelog($dir)
+  {
+    $dir = $dir."nextrelease/";
+    $changelogEntries = [];
+    if ( !is_dir($dir) || !$dh = opendir($dir) ) {
+        throw new \Exception("nextrelease directory doesn't exists or is not readable at location $dir");
+    } else {
+      //Ignore any files starting with a (.) dot 
+      $files=preg_grep('/^([^.])/', scandir($dir));
+      if ( empty($files) ) {
+        throw new \Exception("No release notes files found in $dir folder");
+      }
+      foreach ( $files as $file ) {
+        $str = file_get_contents($dir."/".$file);
+        $changelogEntries= array_merge($changelogEntries, $this->cleanJSON(json_decode($str)));
+      }
+      closedir($dh);
+    }
+    return $changelogEntries;
```

Comment:
If the read changelog is only called once across all those methods, you can have it as a variable there. Yet if it will be called for multiple methods, I'd recommend having a field for it.
Of course it's up to you to take my suggestions or not.